libopenage/renderer/camera/camera.h

```diff
@@ -96,6 +106,19 @@ class Camera {
 	 */
 	void move_rel(Eigen::Vector3f direction, float delta = 1.0f);
 
+
+	/**
+	 * Move the camera position in the direction of a given vector taking the
+	 * camera boundaries into account.
+	 *
+	 * @param direction Direction vector. Added to the current position.
+	 * @param delta Delta for controlling the amount by which the camera is moved. The
+	 *              value is multiplied with the directional vector before its applied to
+	 *              the positional vector.
+	 * @param camera_boundaries X and Z boundaries for the camera in the scene.
+	 */
+	void move_rel(Eigen::Vector3f direction, float delta, CameraBoundaries camera_boundaries);
```

Comment:
Instead of overloading the declaration for `move_rel`, we should probably add `camera_boundaries` to the original `move_rel` implementation and default-initialize it. Then we would always use camera boundaries but the code is a bit simpler. 

The default camera boundaries can be put in `definitions.h` and should use `std::numeric_limits::min()`/`std::numeric_limits::max()` for assignment to the `CameraBoundaries` object. 

```cpp
static constexpr CameraBoundaries DEFAULT_CAM_BOUNDARIES{
    std::numeric_limits::min(),
    std::numeric_limits::max(),
    std::numeric_limits::min(),
    std::numeric_limits::max()
};
```

By default, `std::clamp` would then do practically nothing.

Reply:
I see, makes sense. 

Do you have any proposition of how I should handle the cyclic dependency that arises when the `definitions.h` file needs to access the definition of the `CameraBoundaries` struct?

One way could be to move the definition of the struct to the `definitions.h` file, but I think it feels slightly weird to define it there.

Reply:
For plain data storage structs like this (and enums and special types), we usually put them in a `types.h` file in the same folder. You could also put them into a `boundaries.h` file if you think that makes more sense. Then you can import the file in `camera.h` and `manager.h`.